client/src/main/java/io/split/client/SplitClient.java

```diff
@@ -92,6 +93,49 @@
      */
     String getTreatment(Key key, String split, Map<String, Object> attributes);
 
+    /**
+     * Same as {@link #getTreatment(String, String)} but it returns the configuration associated to the
+     * matching treatment if any. Otherwise {@link SplitResult.configurations()} will be null.
+     * <p/>
+     * <p/>
+     * Examples include showing a different treatment to users on trial plan
+     * vs. premium plan. Another example is to show a different treatment
+     * to users created after a certain date.
+     *
+     * @param key         a unique key of your customer (e.g. user_id, user_email, account_id, etc.) MUST not be null.
+     * @param split    the feature we want to evaluate. MUST NOT be null.
+     * @return the evaluated treatment, the default treatment of this feature, or 'control'.
+     */
+    SplitResult getTreatmentWithConfig(String key, String split);
+
+    /**
+     * Same as {@link #getTreatment(String, String, Map)} but it returns the configuration associated to the
+     * matching treatment if any. Otherwise {@link SplitResult.configurations()} will be null.
+     * <p/>
+     * <p/>
+     * Examples include showing a different treatment to users on trial plan
```

Comment:
An example usage here would be good - like a code snippet with the intended usage pattern